### PR TITLE
Requested python dependencies

### PR DESCRIPTION
**Pull request by maxeaubrey**

### CODE REVIEW

File: dev-python/asgi-redis/asgi-redis-1.4.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+PYTHON_COMPAT=( python{2_7,3_4,3_5,3_6} )
+inherit distutils-r1
+
+DESCRIPTION="Redis-backed ASGI channel layer implementation"
+HOMEPAGE="https://pypi.org/project/asgi-redis/"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="amd64 x86"
```
Comment (alip) [inline dev-python/asgi-redis/asgi-redis-1.4.3.ebuild]: Refrain from using stable keywords unless there's a good reason for it.

File: dev-python/asgi-redis/asgi-redis-1.4.3.ebuild [hunk 2]
```diff
@@ -0,0 +1,23 @@
+EAPI=6
```
Comment (alip) [inline dev-python/asgi-redis/asgi-redis-1.4.3.ebuild]: Missing copyright clause.

File: dev-python/asgiref/asgiref-1.1.2.ebuild [hunk 3]
```diff
@@ -0,0 +1,20 @@
+EAPI=6
```
Comment (alip) [inline dev-python/asgiref/asgiref-1.1.2.ebuild]: ditto.

File: dev-python/asgiref/asgiref-1.1.2.ebuild [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+PYTHON_COMPAT=( python{2_7,3_4,3_5,3_6} )
+inherit distutils-r1
+
+DESCRIPTION="ASGI specs, helper code, and adapters"
+HOMEPAGE="https://pypi.org/project/asgiref/"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="amd64 x86"
```
Comment (alip) [inline dev-python/asgiref/asgiref-1.1.2.ebuild]: ditto.

File: dev-python/boto3/boto3-1.4.4.ebuild [hunk 5]
```diff
@@ -0,0 +1,53 @@
+# Copyright 1999-2018 Gentoo Foundation
```
Comment (alip) [inline dev-python/boto3/boto3-1.4.4.ebuild]: is 2019.

File: dev-python/csscompressor/csscompressor-0.9.4.ebuild [hunk 6]
```diff
@@ -0,0 +1,17 @@
+EAPI=6
```
Comment (alip) [inline dev-python/csscompressor/csscompressor-0.9.4.ebuild]: Missing copyright clause.

File: sci-libs/scikits_learn/scikits_learn-0.20.2.ebuild [hunk 7]
```diff
@@ -0,0 +1,107 @@
+# Copyright 1999-2017 Gentoo Foundation
```
Comment (alip) [inline sci-libs/scikits_learn/scikits_learn-0.20.2.ebuild]: is 2019.

File: sci-libs/scikits_learn/scikits_learn-0.20.2.ebuild [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+	# bug #397605
+	[[ ${CHOST} == *-darwin* ]] \
+		&& append-ldflags -bundle "-undefined dynamic_lookup" \
+		|| append-ldflags -shared
+
+	# scikits-learn now uses the horrible numpy.distutils automagic
+	export SCIPY_FCONFIG="config_fc --noopt --noarch"
+
+	# remove bundled cblas
+	rm -r sklearn/src || die
```
Comment (alip) [inline sci-libs/scikits_learn/scikits_learn-0.20.2.ebuild]: Looks like this needs `-rf`.
Add a short diagnostics message to die such as `die "failed to remove bundled cblas"`.

File: sci-libs/scikits_learn/scikits_learn-0.20.2.ebuild [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+PATCHES=( "${FILESDIR}"/${PN}-0.18.1-system-cblas.patch )
+
+python_prepare_all() {
+	# bug #397605
+	[[ ${CHOST} == *-darwin* ]] \
+		&& append-ldflags -bundle "-undefined dynamic_lookup" \
+		|| append-ldflags -shared
+
+	# scikits-learn now uses the horrible numpy.distutils automagic
+	export SCIPY_FCONFIG="config_fc --noopt --noarch"
```
Comment (alip) [inline sci-libs/scikits_learn/scikits_learn-0.20.2.ebuild]: Common practice is to define this in global scope. There's no direct guarantee that a variable exported in a phase will be available to subsequent phases.

File: sci-libs/scikits_learn/scikits_learn-0.20.2.ebuild [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+
+python_test() {
+	# doc builds and runs tests
+	use doc && return
+	distutils_install_for_testing ${SCIPY_FCONFIG}
+	esetup.py install \
+			  --root="${T}/test-${EPYTHON}" \
+			  --no-compile ${SCIPY_FCONFIG}
+	pushd "${T}/test-${EPYTHON}/$(python_get_sitedir)" || die > /dev/null
+	JOBLIB_MULTIPROCESSING=2 SKLEARN_SKIP_NETWORK_TESTS=1 nosetests -v sklearn --exe || die
```
Comment (alip) [inline sci-libs/scikits_learn/scikits_learn-0.20.2.ebuild]: Short diagnostics message for die is common practice.

File: sci-libs/scikits_learn/scikits_learn-0.20.2.ebuild [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+	popd > /dev/null
+}
+
+python_install() {
+	distutils-r1_python_install ${SCIPY_FCONFIG}
+}
+
+python_install_all() {
+	find "${S}" -name \*LICENSE.txt -delete
```
Comment (alip) [inline sci-libs/scikits_learn/scikits_learn-0.20.2.ebuild]: Common practice is to do this in `src_prepare()`.

File: dev-python/csscompressor/csscompressor-0.9.4.ebuild [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+PYTHON_COMPAT=( python{2_7,3_4,3_5,3_6} )
+inherit distutils-r1
+
+DESCRIPTION="A python port of YUI CSS Compressor"
+HOMEPAGE="https://pypi.org/project/csscompressor/"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="amd64 x86"
```
Comment (alip) [inline dev-python/csscompressor/csscompressor-0.9.4.ebuild]: Refrain from using stable keywords unless there's a good reason for it.

File: sci-libs/scikits_learn/scikits_learn-0.20.2.ebuild [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+	distutils-r1_python_install_all
+	if use examples; then
+		dodoc -r examples
+		docompress -x /usr/share/doc/${PF}/examples
+	fi
+
+}
+
```
Comment (alip) [inline sci-libs/scikits_learn/scikits_learn-0.20.2.ebuild]: Nitpick: Useless blank line on last line.
Use `repoman full` ;)

File: dev-python/django-postgres-copy/django-postgres-copy-0.2.2.ebuild [hunk 14]
```diff
@@ -0,0 +1,20 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python{2_7,3_4,3_5,3_6} )
+inherit distutils-r1
+
+DESCRIPTION="Quickly import and export delimited data with Django support for PostgreSQL&#39;s COPY command"
```
Comment (patricklauer) [inline dev-python/django-postgres-copy/django-postgres-copy-0.2.2.ebuild]: PostgreSQL&#39;s :D

File: dev-python/social-auth-core/social-auth-core-1.6.0.ebuild [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+DEPEND="
+	dev-python/setuptools[${PYTHON_USEDEP}]
+"
+RDEPEND="
+	dev-python/requests[${PYTHON_USEDEP}]
+	dev-python/oauthlib[${PYTHON_USEDEP}]
+	dev-python/requests-oauthlib[${PYTHON_USEDEP}]
+	dev-python/six[${PYTHON_USEDEP}]
+	dev-python/pyjwt[${PYTHON_USEDEP}]
+	python_targets_python2_7? ( dev-python/python-openid[python2_7] )
```
Comment (patricklauer) [inline dev-python/social-auth-core/social-auth-core-1.6.0.ebuild]: `dev-python/python-openid[${PYTHON_USEDEP}]` catches all variations of this

File: dev-python/django-postgres-copy/django-postgres-copy-0.2.2.ebuild (hunk 14 above)
Comment (maxeaubrey) [reply dev-python/django-postgres-copy/django-postgres-copy-0.2.2.ebuild]: i just curl what the pypi description says :)

**Comment by maxeaubrey** [comment]
Thanks to @alip for finding and helping me understand that last part :)

**Comment by patricklauer** [comment]
Looks good to me.

**Pull request merged**